hammerblade/torch/tests/test_cat.py

```diff
@@ -3,60 +3,71 @@
 """
 import torch
 import pytest
-def _test_torch_cat(x):
+from hypothesis import given, settings
+from .hypothesis_test_util import HypothesisUtil as hu
+
+def _test_torch_cat(x, y, z):
     x_h = x.hammerblade()
-    y = torch.cat([x, x, x], 0)
-    y_h = torch.cat([x_h, x_h, x_h], 0)
+    y_h = y.hammerblade()
+    z_h = z.hammerblade()
+    a = torch.cat([x, y, z], 0)
+    a_h = torch.cat([x_h, y_h, z_h], 0)
     assert y_h.device == torch.device("hammerblade")
     assert torch.allclose(y, y_h.cpu())
 
 def test_cat_1():
     x = torch.ones(10)
-    _test_torch_cat(x)
+    _test_torch_cat(x, x, x)
 
 def test_cat_1_dif_sizes():
     x = torch.randn(3)
     y = torch.randn(2)
     z = torch.tensor([])
-    x_h = x.hammerblade()
-    y_h = y.hammerblade()
-    z_h = z.hammerblade()
-    a = torch.cat([x, y, z], 0)
-    a_h = torch.cat([x_h, y_h, z_h], 0)
-    assert a_h.device == torch.device("hammerblade")
-    assert torch.allclose(a, a_h.cpu())
+    _test_torch_cat(x, y, z)
+
+@settings(deadline=None)
+@given(inputs=hu.tensors1d(n=3))
+def test_cat_1_hypothesis(inputs):
+    x1 = torch.tensor(inputs[0])
+    x2 = torch.tensor(inputs[1])
+    x3 = torch.tensor(inputs[2])
+    _test_torch_cat(x1, x2, x3)
 
 def test_cat_2():
     x = torch.randn(3, 4)
-    _test_torch_cat(x)
+    _test_torch_cat(x, x, x)
 
 def test_cat_2_dif_sizes():
     x = torch.randn(3, 4)
     y = torch.randn(2, 4)
     z = torch.randn(4, 4)
-    x_h = x.hammerblade()
-    y_h = y.hammerblade()
-    z_h = z.hammerblade()
-    a = torch.cat([x, y, z], 0)
-    a_h = torch.cat([x_h, y_h, z_h], 0)
-    assert a_h.device == torch.device("hammerblade")
-    assert torch.allclose(a, a_h.cpu())
+    _test_torch_cat(x, y, z)
+
+@settings(deadline=None)
+@given(inputs=hu.tensors(n=3, min_dim=2, max_dim=2))
```

Comment:
Awesome! You figured out how to use our (ugly) hypothesis wrappers!

One more tip -- you dont have to have 2 cases, one for 2D and one for 3D. Hypothesis will do it for you. `hu.tensors` will only generate tensors with the same dimensionality